.github/workflows/tests_talk2aiagents4pharma.yml

```diff
@@ -0,0 +1,400 @@
+# This is a basic workflow to help you get started with GitHub Actions
+name: TESTS Talk2AIAgents4Pharma
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request events
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'aiagents4pharma/talk2aiagents4pharma/**'
+      - 'aiagents4pharma/talk2biomodels/**'
+      - 'aiagents4pharma/talk2knowledgegraphs/**'
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+env:
+  OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+  NVIDIA_API_KEY: ${{ secrets.NVIDIA_API_KEY }}
+
+# This workflow contains jobs covering linting and code coverage (along with testing).
+jobs:
+  # pylint job for macos
+  pylint-macos:
+    # The type of runner that the job will run on
+    name: pylint
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+        matrix:
+            os: [macos-13]
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.12
+
+      - name: Cache files
+        uses: actions/cache@v4
+        id: cache
+        with:
+          path: |
+            ${{ github.workspace }}/venv/*
+          key: ${{ runner.os }}-venv
+
+      # install requirements
+      - name: Install the requirements
+        if: steps.cache.outputs.cache-hit != 'true'
+        working-directory: ${{ github.workspace }}
+        run: |
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install --upgrade pip
+          pip3 install --break-system-packages -r requirements.txt
+
+      # pylint
+      - name: Run pylint
+        working-directory: ${{ github.workspace }}
+        run: |
+          source venv/bin/activate
+          pylint --disable=R0801,R0902,W0221,W0122 aiagents4pharma/talk2aiagents4pharma
```

Comment:
Nice work adapting this workflow to conditionally run on changes to Talk2AIAgents4Pharma, T2B, or T2KG and then just running the tests for Talk2AiAgents4Pharma.